File: examples/python-vue/backend/services/todos_service.py
import uuid
from generated.interfaces.i_todos_service import ITodosService

_store = [
    {"id": "1", "title": "Buy groceries", "completed": False, "userId": "1"},
    {"id": "2", "title": "Read docs",     "completed": True,  "userId": "2"},
]


class TodosService(ITodosService):
    def list_todos(self):
        return list(_store)

    def get_todo(self, id: str):
        todo = next((t for t in _store if t["id"] == id), None)
        if todo is None:
            raise LookupError(f"Todo {id} not found")
        return todo

    def create_todo(self, input):
        todo = {"id": str(uuid.uuid4()), "completed": False, **input}
        _store.append(todo)
        return todo

    def update_todo(self, id: str, input):
        todo = next((t for t in _store if t["id"] == id), None)
        if todo is None:
            raise LookupError(f"Todo {id} not found")
        if input.get("title") is not None:
            todo["title"] = input["title"]
        if input.get("completed") is not None:
            todo["completed"] = input["completed"]
        return todo

    def delete_todo(self, id: str):
        idx = next((i for i, t in enumerate(_store) if t["id"] == id), None)
        if idx is None:
            raise LookupError(f"Todo {id} not found")
        _store.pop(idx)

File: examples/node-react/backend/src/services/TodosService.ts
import { ITodosService } from "../../../generated/interfaces/ITodosService";
import { Todo, CreateTodoInput, UpdateTodoInput } from "../../../generated/types/todos";
import { todosErrors } from "../../../generated/errors/todos.errors";
import { randomUUID } from "crypto";

const store: Todo[] = [
  { id: "1", title: "Buy groceries", completed: false, userId: "1" },
  { id: "2", title: "Read docs",     completed: true,  userId: "2" },
];

export class TodosService implements ITodosService {
  async listTodos(): Promise<Todo[]> {
    return store;
  }

  async getTodo(id: string): Promise<Todo> {
    const todo = store.find((t) => t.id === id);
    if (!todo) throw todosErrors.getTodo.notFound(`Todo ${id} not found`);
    return todo;
  }

  async createTodo(input: CreateTodoInput): Promise<Todo> {
    if (!input.title) throw todosErrors.createTodo.badRequest("Title is required");
    const todo: Todo = { id: randomUUID(), completed: false, ...input };
    store.push(todo);
    return todo;
  }

  async updateTodo(id: string, input: UpdateTodoInput): Promise<Todo> {
    const todo = store.find((t) => t.id === id);
    if (!todo) throw todosErrors.updateTodo.notFound(`Todo ${id} not found`);
    if (!input.title && input.completed === undefined) {
      throw todosErrors.updateTodo.badRequest("At least one field must be provided");
    }
    if (input.title     !== undefined) todo.title     = input.title;
    if (input.completed !== undefined) todo.completed = input.completed;
    return todo;
  }

  async deleteTodo(id: string): Promise<void> {
    const idx = store.findIndex((t) => t.id === id);
    if (idx === -1) throw todosErrors.deleteTodo.notFound(`Todo ${id} not found`);
    store.splice(idx, 1);
  }
}
